ttnn/cpp/ttnn/types.hpp

```diff
@@ -123,11 +121,12 @@ class Buffer : public tt::tt_metal::Buffer {
             }
             this->set_size(0);
             TT_ASSERT(this->device()->allocator_ != nullptr, "Expected allocator to be initialized!");
-            // Extract the required buffer attributes from main thread (these are guaranteed to be correctly populated) and send to worker
-            this->device()->push_work([dev = this->device(), id = this->buffer_id, type = this->buffer_type()] () mutable {
+            // Extract the required buffer attributes from main thread (these are guaranteed to be correctly populated)
+            // and send to worker Change here can be ignored since ttnn buffer is going away
+            this->device()->push_work([dev = this->device(), buffer = this]() mutable {
                 // At this point, the address for this buffer has made it to GLOBAL_BUFFER_ADDRESS_MAP, since the worker has allocated the buffer.
-                tt::tt_metal::allocator::deallocate_buffer(*(dev->allocator_), GLOBAL_BUFFER_ADDRESS_MAP.buffer_address(id), type);
-                GLOBAL_BUFFER_ADDRESS_MAP.erase(id);
+                tt::tt_metal::allocator::deallocate_buffer(*(dev->allocator_), buffer);
+                GLOBAL_BUFFER_ADDRESS_MAP.erase(buffer->buffer_id);
```

Comment:
What is the GLOBAL_BUFFER_ADDRESS_MAP can we remove it too?

Reply:
Yes, that is being removed in #14023 which this pr has a dependency on, so this pr is currently in draft. The entire ttnn buffer class is being removed so once the other pr is merged there won't be any changes in this pr to this file. Change here is just to make sure my changes compile before rebasing.